src/Compilers/CSharp/Test/Symbol/Symbols/Metadata/PE/LoadCustomModifiers.cs

```diff
@@ -250,6 +250,1915 @@ public void TestFieldCustomModifierCount()
+        }
+
+        [Fact]
+        public void MissingInAttributeModreq_Properties_ModOpt()
+        {
+            var reference = CompileIL(@"
+.class public auto ansi beforefieldinit RefTest extends [mscorlib]System.Object
+{
+    .method public hidebysig specialname instance int32& modopt([mscorlib]System.Runtime.InteropServices.InAttribute) get_X () cil managed 
+    {
+        .param [0]
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = (01 00 00 00)
+        .maxstack 8
+
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Object::.ctor()
+        IL_0006: ret
+    }
+
+    .property instance int32& modopt([mscorlib]System.Runtime.InteropServices.InAttribute) X()
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = (01 00 00 00)
+        .get instance int32& modopt([mscorlib]System.Runtime.InteropServices.InAttribute) RefTest::get_X()
+    }
+}");
+
+            CreateStandardCompilation(@"
+class Test
+{
+    public ref readonly int M(RefTest obj) => ref obj.X;
+}", references: new[] { reference }).VerifyDiagnostics(
+                // (4,55): error CS0570: 'RefTest.X' is not supported by the language
+                //     public ref readonly int M(RefTest obj) => ref obj.X;
+                Diagnostic(ErrorCode.ERR_BindToBogus, "X").WithArguments("RefTest.X").WithLocation(4, 55));
+        }
+
+        [Fact]
+        public void MissingInAttributeModreq_Method_Parameters_Virtual()
+        {
+            var reference = CompileIL(@"
+.class public auto ansi beforefieldinit RefTest extends [mscorlib]System.Object
+{
+    .method public hidebysig newslot virtual instance void M ([in] int32& x) cil managed 
+    {
+        .param [1]
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = (01 00 00 00)
+        .maxstack 8
+
+        IL_0000: ret
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Object::.ctor()
+        IL_0006: ret
+    }
+}");
+
+            CreateStandardCompilation(@"
+class Test
+{
+    public int M(RefTest obj) => obj.M(0);
+}", references: new[] { reference }).VerifyDiagnostics(
+                // (4,38): error CS0570: 'RefTest.M(in int)' is not supported by the language
+                //     public int M(RefTest obj) => obj.M(0);
+                Diagnostic(ErrorCode.ERR_BindToBogus, "M").WithArguments("RefTest.M(in int)").WithLocation(4, 38));
+        }
+
+        [Fact]
+        public void MissingInAttributeModreq_Method_Parameters_Virtual_ModOpt()
```

Comment:
Would it be sufficient to have one `_ModOpt` version of the tests for methods and one for properties. 

Also, some of the suffixes are `_Modopt` and some are `_ModOpt`.

Reply:
@cston can you please clarify?
fixed the naming issue.

---
In reply to: [167270920](https://github.com/dotnet/roslyn/pull/24524#discussion_r167270920) [](ancestors = 167270920)

Reply:
There are twenty pairs of tests where each pair has one test with missing attribute and one test with `modopt` rather than `modreq`. That is a large number of tests. But `HasInAttributeModifier` checks for both the attribute and `modreq` so it seems like overkill to have two tests for each. It feels like one or two tests for `modopt` should be sufficient.

Reply:
Is there a test where both `modopt` and `modreq` are included?

Reply:
There are tests where a virtual method in a parent class has a modreq, but the overriding method in a child class has a modopt. I'm adding the opposite right now.

---
In reply to: [167356516](https://github.com/dotnet/roslyn/pull/24524#discussion_r167356516) [](ancestors = 167356516)